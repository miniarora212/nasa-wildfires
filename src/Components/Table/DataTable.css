.TableContainer{
    display: flex;
    justify-content: flexStart;
    flex-direction: column;
    padding-bottom:1rem;
}
.TableContainer thead.ant-table-thead th {
    background: #26545c;
    color: #fff;
}
.TableContainer tbody tr:nth-child(even) {
    background: rgb(38 84 92 / 19%);
}
.TableContainer .ant-table-cell a {
    background: #26545c;
    color: #fff;
    padding: 8px 15px;
    border-radius: 5px;
    font-weight: bold;
}
.ant-pagination-item-link {
    background: #26545c !important;
    color: #fff !important;
    font-weight: bold;
}