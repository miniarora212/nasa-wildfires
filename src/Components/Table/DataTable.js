import React from 'react'
import * as PropTypes from 'prop-types';
import Loader from '../Loader/Loader'
import { Table } from 'antd';
import './DataTable.css'
const DataTable = (props) => {
    const {tableData, isLoading} = props;
    const pageSizes = [5, 10, 20, 50] // Page size for the Pagination
    const defaultPageSize = 5;

    return (
        <>
            {
                isLoading ? <Loader />
                    :
                    <div className="TableContainer mt-5">
                        <h2 className="table-heading">
                            Wildfires Data
                        </h2>
                        <Table
                            dataSource={tableData}
                            pagination={tableData.length <= 5 ? false : { defaultPageSize: defaultPageSize, showSizeChanger: true, pageSizeOptions:pageSizes }}
                            columns={[{
                                title: "ID", dataIndex: "id", key: "id"
                            },
                            {
                                title: "Start Date", dataIndex: "startdate", key: "startdate"
                            }, {
                                title: "Closed Date", dataIndex: "closeddate", key: "closeddate"
                            }, {
                                title: "Title", dataIndex: "title", key: "title"
                            }, {
                                // eslint-disable-next-line
                                title: "Source Info", dataIndex: "inciweb", key: "inciweb", render: text => <a target='_blank' href={text}>View</a>,
                            },]}
                            rowKey='id'
                        />;
                    </div>
            }
        </>
    )
}

DataTable.propTypes = {
    isLoading: PropTypes.bool.isRequired,
    tableData: PropTypes.array.isRequired,
  }

export default DataTable
