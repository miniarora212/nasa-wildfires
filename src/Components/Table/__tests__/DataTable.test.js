import { cleanup, render } from "@testing-library/react";
import React from "react"
import DataTable from "../DataTable";

afterEach(cleanup);
window.matchMedia = window.matchMedia || function() {
    return {
        matches: false,
        addListener: function() {},
        removeListener: function() {}
    };
};

describe("DataTable", () => {
    const defaultProps = {
        isLoading: false,
        tableData: [{id:'1', startDate:"2021-08-12", closeDate:"2021-11-02", title:"Test fire", inciweb:"https://test.com"}]
    };

    it("Should render the Data table successfully", () => {
        const { asFragment } = render(<DataTable {...defaultProps} />);
        expect(asFragment(<DataTable {...defaultProps}/>)).toMatchSnapshot();
    })
})