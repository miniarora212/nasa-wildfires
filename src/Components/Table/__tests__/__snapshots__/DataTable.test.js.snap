// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable Should render the Data table successfully 1`] = `
<DocumentFragment>
  <div
    class="TableContainer mt-5"
  >
    <h2
      class="table-heading"
    >
      Wildfires Data
    </h2>
    <div
      class="ant-table-wrapper"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: auto;"
                >
                  <colgroup />
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell"
                      >
                        ID
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Start Date
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Closed Date
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Title
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Source Info
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        1
                      </td>
                      <td
                        class="ant-table-cell"
                      />
                      <td
                        class="ant-table-cell"
                      />
                      <td
                        class="ant-table-cell"
                      >
                        Test fire
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a
                          href="https://test.com"
                          target="_blank"
                        >
                          View
                        </a>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    ;
  </div>
</DocumentFragment>
`;
